src/main/java/com/ldtteam/structurize/blueprints/v1/Blueprint.java

```diff
@@ -948,7 +953,15 @@ public boolean equals(final Object obj)
             return false;
         }
         final Blueprint other = (Blueprint) obj;
-        return Objects.equals(name, other.name) && palleteSize == other.palleteSize && getVolume() == other.getVolume() && Objects.equals(filePath, other.filePath);
+        return Objects.equals(name, other.name)
+                 && Objects.equals(fileName, other.fileName)
+                 && Objects.equals(filePath, other.filePath)
+                 && Objects.equals(packName, other.packName)
+                 && Objects.equals(rotation, other.rotation)
+                 && Objects.equals(mirror, other.mirror)
+                 && palleteSize == other.palleteSize
+                 && getVolume() == other.getVolume()
+                 && Objects.equals(renderSource, other.renderSource);
```

Comment:
What is render source, this hash/equality controls render cache iirc, so it should ideally not create multiple buffers for same data

Reply:
render source is the display position

Reply:
Oh yeah, dont include the position then in here. We want to maximize caching.

Reply:
But equal hashcode will just display the same thing, even though its not the same thing. It shouldnt constantly switch, right?

Reply:
(One of the players will see the wrong blueprint)

Reply:
they're calling update on each draw, after retriving the renderer via hashcode, so its possible that two handlers with different blueprints but matching hashcodes constantly re-inject themselves into the same renderer

Reply:
One of the laws of hashcodes is that any two things that are `equals` must have the same hash.  The hash is allowed to be weaker (though that needs justification -- "the field is mutable" is a valid justification though) but is not allowed to be stronger.